### ISSUE TITLE
Failed to fine <sic> file matching any known edition

### ISSUE THREAD

**Issue by richgain** [IHTSDO/rf2-to-rf1-conversion#1]
I get the following output:
```
C:\Java\rf2-to-rf1-conversion-master>go
How much memory do you have available? [10g]:
Do you have a 2nd drive? (eg C:\) Y/N: y
Where is the RF2 Archive? [D:\Coding\Resources\SnomedCT_USEditionRF2_Production_20170301T120000.zip]:

C:\Java\rf2-to-rf1-conversion-master>java -jar -Xms4g -Xmx10g  target\RF2toRF1Converter.jar -u C:\  D:\Coding\Resources\SnomedCT_USEditionRF2_Production_20170301T120000.zip
Initialising Database
Creating database schema
10.61% complete.
Extracting RF2 International Edition Data...
14.67% complete.
Process failed in 18.93 s after completing 65/443 operations.

Cleaning up resources...
15.12% complete.Exception in thread "main" org.ihtsdo.snomed.rf2torf1conversion.RF1ConversionException: Failed to fine file matching any known edition: sct2_Description_EXTFull-LNG_INT_DATE.txt inC:\rf2-to-rf1-987156874018252316
        at org.ihtsdo.snomed.rf2torf1conversion.ConversionManager.determineEdition(ConversionManager.java:429)
        at org.ihtsdo.snomed.rf2torf1conversion.ConversionManager.doRf2toRf1Conversion(ConversionManager.java:210)
        at org.ihtsdo.snomed.rf2torf1conversion.ConversionManager.main(ConversionManager.java:193)

```
Consequently, I am currently unable to create an RF1 version of the US version of SNOMED.

**Comment by pgwilliams** [comment]
Hi Richgain, thanks for spotting this issue.   As far as I'm aware this is the first time that anyone has tried the RF2 -> RF1 conversion with the US Edition.  It was developed to target just the International and Spanish editions.  So while it does have some support for extensions, we've not really developed it in that direction.  Apologies for not making that limitation explicit in the Readme file.

I'll put some time in to this early next week and hopefully get something working for you.  
Best Wishes,
Peter

**Comment by richgain** [comment]
Thanks Peter.
Sadly, the US seem to be taking the RF2 deadline as a hard stop. No RF1 version this release and no mention of the converter on the NLM site either.

**Comment by richgain** [comment]
Hi Peter
Any luck with fixing the conversion tool for US extension files?
Thanks,
Richard

**Comment by pgwilliams** [comment]
Hi Richard.  Yes!  Making the conversion tool recognise the US edition was a quick job, I just needed to add the module id.  However, testing uncovered some data issues with the US Edition (some known, some news to us) which I needed to write some workarounds/cleanup for.     The code is now working through to completion, but really I want to compare a conversion of March 2016 done using to the tool against what the NLM themselves released and also a regression test to ensure I haven't impacted the International Edition.    I can either check the code in now, or once testing is complete - whatever you prefer?

**Comment by richgain** [comment]
Hi
That sounds positive. Would you mind checking the code in now so I can
start experimenting with the output?
I will certainly await the outcome of your testing before building our new
release though.
Thanks,
Richard

**Comment by pgwilliams** [comment]
Done!  Please ensure that you're working from the develop branch.  Obviously I don't want to merge into master before the testing is complete.   Can I ask, out of interest, who you're creating this package for, Richard?  Feel free to email me directly for privacy.   pwi at snomed . org

**Comment by pgwilliams** [comment]
I used the following command to process the US Edition:

java -jar -Xmx8g target/RF2toRF1Converter.jar ~/tmp/us_edition/SnomedCT_USEditionRF2_Production_20170301T120000.zip -v -u /Volumes/ram_disk

If you've got the memory to support it, a Ram Disk does speed up the initial stages.

**Comment by rakhmatbudi** [comment]
hi.
I tried to run the conversion program against the bundle that only consist of snapshot. And it failed.
Any plan to release version that support snapshot?

**Comment by pgwilliams** [comment]
Hi Rakhmat - as discussed, no current plans.   You're the first person to mention it!   Feel free to create a new issue to request this feature.  Issue #1 should have been marked as closed.
